front/src/applications/editor/tools/rangeEdition/speedSection/SpeedSectionEditionLayers.tsx

```diff
@@ -72,17 +75,29 @@ export const SpeedSectionEditionLayers = () => {
     return res;
   }, [interactionState, hoveredItem, entity, selectedSwitches]);
 
-  const highlightedTracksFeature: FeatureCollection = useMemo(() => {
+  const makeTrackRangeFeatures =
+    (flatEntity: EditorEntity) => (range: ApplicableTrackRange, i: number) => {
+      const trackState = trackSectionsCache[range.track];
+      return trackState?.type === 'success'
+        ? getTrackRangeFeatures(trackState.track, range, i, flatEntity.properties)
+        : [];
```

Comment:
Cast here to avoid casting 3 times below
```suggestion
      return (
        trackState?.type === 'success'
          ? getTrackRangeFeatures(trackState.track, range, i, flatEntity.properties)
          : []
      ) as Feature<LineString | Point>[];
```

Reply:
nicely spotted